ESP/lib/src/network/wifihandler/wifihandler.cpp

```diff
@@ -169,6 +171,20 @@ bool WiFiHandler::iniSTA(const std::string& ssid,
     currentMillis = millis();
     log_i(".");
     log_d("Progress: %d \n\r", progress);
+    
+    // Check if mode has been changed to USB mode during connection attempt
+    DeviceModeManager* deviceModeManager = DeviceModeManager::getInstance();
+    if (deviceModeManager && deviceModeManager->getMode() == DeviceMode::USB_MODE) {
+      log_i("[WiFiHandler] Mode changed to USB during connection, aborting WiFi setup");
+      WiFi.disconnect(true);
+      wifiStateManager.setState(WiFiState_e::WiFiState_Disconnected);
+      return false;
+    }
+    
+    if (Serial.available()) {
+      yield();
+    }
```

Comment:
So, the way I understand it. We're connecting to wifi, but then a command came telling us to use usb instead, so we've changed the mode. Why do we have to yield here then? Shouldn't wifi manager just stop in its track once we disconnect?